src/NuGet.Core/NuGet.Commands/RestoreCommand/Utility/LockFileUtils.cs

```diff
@@ -266,38 +270,152 @@ public static class LockFileUtils
+                // Retrieve the resolved framework name, if this is null it means that the
+                // project is incompatible. This is marked as Unsupported.
+                var targetFrameworkInformation = (TargetFrameworkInformation)frameworkInfoObject;
+                projectFramework = targetFrameworkInformation.FrameworkName?.DotNetFrameworkName
+                    ?? NuGetFramework.UnsupportedFramework.DotNetFrameworkName;
             }
 
-            if ((dependencyType & LibraryIncludeFlags.Compile) == LibraryIncludeFlags.None)
+            // Create the target entry
+            var projectLib = new LockFileTargetLibrary()
             {
-                ClearIfExists(lockFileLib.CompileTimeAssemblies);
-            }
+                Name = library.Name,
+                Version = library.Version,
+                Type = LibraryType.Project,
+                Framework = projectFramework,
+
+                // Find all dependencies which would be in the nuspec
+                // Include dependencies with no constraints, or package/project/external
+                // Exclude suppressed dependencies, the top level project is not written 
+                // as a target so the node depth does not matter.
+                Dependencies = graphItem.Data.Dependencies
+                    .Where(
+                        d => (d.LibraryRange.TypeConstraintAllowsAnyOf(
+                            LibraryDependencyTarget.PackageProjectExternal))
+                             && d.SuppressParent != LibraryIncludeFlags.All)
+                    .Select(d => GetDependencyVersionRange(d))
+                    .ToList()
+            };
 
-            if ((dependencyType & LibraryIncludeFlags.Native) == LibraryIncludeFlags.None)
+            if (rootProjectStyle == ProjectStyle.PackageReference)
             {
-                ClearIfExists(lockFileLib.NativeLibraries);
+                // Add files under asset groups
+                object filesObject;
+                object msbuildPath;
+                if (localMatch.LocalLibrary.Items.TryGetValue(KnownLibraryProperties.ProjectRestoreMetadataFiles, out filesObject)
+                     && localMatch.LocalLibrary.Items.TryGetValue(KnownLibraryProperties.MSBuildProjectPath, out msbuildPath))
+                {
+                    var files = (List<ProjectRestoreMetadataFile>)filesObject;
+                    var fileLookup = new Dictionary<string, ProjectRestoreMetadataFile>(StringComparer.OrdinalIgnoreCase);
+
+                    foreach (var file in files)
+                    {
+                        var path = file.PackagePath;
+
+                        // LIBANY avoid compatibility checks and will always be used.
+                        if (LIBANY.Equals(path, StringComparison.Ordinal))
+                        {
+                            path = $"lib/{targetGraph.Framework.GetShortFolderName()}/{Guid.NewGuid()}.dll";
```

Comment:
this is kind of the opposite :smile:

``_._`` will tell the build task that this project should not be included in the build (transitive: false). Any other dlls will mean that it should be added, and the build task will figure out the real dll itself. So the code here looks a bit odd, but it is just trying to add a reference to a new fake dll that won't conflict.